src/lib/integration_test_lib/stack_driver_log_engine.ml

```diff
@@ -420,24 +432,20 @@ module Make (Testnet : Test_intf.Testnet_intf) :
                   | `Snarked_ledgers_generated of int
                   | `Milliseconds of int64 ]
       -> t
-      -> unit Deferred.t =
+      -> unit Deferred.Or_error.t =
    fun ?(blocks = 0) ?(epoch_reached = 0) ?(timeout = `Milliseconds 300000L) t ->
-    match%bind
-      Deferred.Or_error.try_with_join (fun _ ->
-          wait_for' ~blocks ~epoch_reached ~timeout t )
-    with
+    match%bind wait_for' ~blocks ~epoch_reached ~timeout t with
     | Ok _ ->
-        Deferred.unit
+        Deferred.Or_error.ok_unit
     | Error e -> (
         Logger.fatal t.logger ~module_:__MODULE__ ~location:__LOC__
           "wait_for failed with error: $error"
           ~metadata:[("error", `String (Error.to_string_hum e))] ;
-        match%map cleanup t with
-        | None ->
-            failwith (Error.to_string_hum e)
-        | Some err_str ->
-            failwith
-              (String.concat ~sep:" and " [Error.to_string_hum e; err_str]) )
+        match%map delete t with
```

Comment:
Nit: This match is redundant, `let%map res = delete t in Or_error.combine_errors_unit [Error e; res]`

Reply:
oops, yes